docs/changes/table/18680.api.rst

```diff
@@ -0,0 +1,5 @@
+Selecting a table index in the ``.loc``, ``.iloc``, or ``.loc_indices`` properties by
+passing the index identifier as the first element of the item is deprecated and is
+planned for removal in astropy 9.0. For example, if a table ``t`` has two indices on
```

Comment:
I don't think changelogs should contain this kind of promises.
```suggestion
passing the index identifier as the first element of the item is deprecated. For example, if a table ``t`` has two indices on
```

Reply:
I *completely* agree about not making promises about future plans (and this is something I notice all the time). 

However to me this is providing useful guidance to developers about the intended timeline. I was also careful to say "planned", which is not a promise. Note there is recent precedent from @mhvk which does make a stronger promise:
```
✗ more docs/changes/coordinates/18193.api.rst
The functionality of ``astropy.coordinates.concatenate`` and
``astropy.coordinates.concatenate_representations`` is now available using
``np.concatenate``. Hence, these functions are being deprecated, emitting an
``AstropyPendingDeprecationWarning`` starting with astropy 7.2. This will be
followed by a regular deprecation warning in astropy 8.0, and removal in 9.0.
```

Reply:
Another way to view this is not a promise but a warning... it's coming by 9.0, so you had better be ready.

Reply:
I missed that. I'll open another PR to propose dropping all such promises then.

Reply:
I'm with @taldcroft here, in that for the changelog it is fine to give these warnings - they are read only at the time of release, almost certainly not by regular users, and give a sense that we're serious about it. I do agree, though, that in the documentation we should not add such statements. Rather, we should simply not use anything deprecated.